source/common/grpc/json_transcoder_filter.cc

```diff
@@ -70,6 +72,14 @@ class TranscoderImpl : public Transcoder {
   std::unique_ptr<TranscoderInputStream> response_stream_;
 };
 
+bool isGrpcResponse(const Http::HeaderMap& headers) {
+  if (strcmp("200", headers.Status()->value().c_str()) != 0) {
+    return false;
+  }
+  return headers.ContentType() &&
+         Http::Headers::get().ContentTypeValues.Grpc == headers.ContentType()->value().c_str();
```

Comment:
I think this will cause a string temporary to be created. If you invert it won't:
` headers.ContentType()->value() == Http::Headers::get().ContentTypeValues.Grpc.c_str()`

Reply:
Done.

I'm fine either though I don't think it will make a string temporary, as `<string>` defines:
```
bool operator== (const string& lhs, const char*   rhs);
```
http://www.cplusplus.com/reference/string/string/operators/

Reply:
Ah, good to know. Thanks for pointing that out.